src/server/services/checkWebsiteStatus.js

```diff
@@ -0,0 +1,52 @@
+const cron = require('node-cron');
+const { getConnection } = require('../db');
+const { logMailer } = require('../logMailer');
+const LogEmail = require('../models/LogEmail');
+const colors = require('colors');
+  
+async function checkWebsite() {
+  const conn = getConnection();
+
+  try {
+    const response = await fetch('https://httpstat.us/500/', { method: 'HEAD' });
+
+    if (response.ok) {
+      console.log('The server is up'.green.inverse);
+    } else {
+      console.error('The server is down'.red.inverse);
+
+      const errorMessage = 'The server at https://httpstat.us/500/ is down.';
+
+      // Log the error in the database
+      const log = new LogEmail(undefined, errorMessage);
+      await log.insert(conn);
+
+      // Send a logging email
+      await logMailer(conn);
+    }
+  } 
+  catch (error) {
+    console.error('Error:', error);
+
+    const errorMessage = `Failed to reach https://httpstat.us/500/. Error: ${error.message}`;
+
+    // Log the error in the database
+    const log = new LogEmail(undefined, errorMessage);
+    await log.insert(conn);
+
+    // Send a logging email
+    await logMailer(conn);
+  } finally {
+    if (conn.close) {
+      await conn.close();
+    }
+  }
+}
+
+// Schedule the task to run every hour
```

Comment:
I like this idea of starting cron inside of OED code. What I'm wondering is how a site would easily run this. It worked fine by doing ``npm run status`` inside an OED Docker web container. However, it stopped if I terminated OED so it does not tell you OED is down when that happens. The only way run outside the Docker container that I know is to require installing node on the server. That is extra work and OED does not want to modify the server software. Do you know how to do this in another way or make sure it does not stop if OED is down? Up to now OED tells sites to set up a cron job via Linux/Unix on their server to run tasks. If this new way was easy to do then OED would consider using it and switching other cron uses to this.

Reply:
You are right that the script will stop running once your terminate the docker container. I initially did it think way in order to make it easy for user to run without having to enter cron commands.

To fix this, there are 2 ways to do it that I can think of:
1. Have user enter the cron command below, it will then run the script regardless docker container is open or not.

crontab -e
0 * * * * /usr/bin/node /absolute/path/to/src/server/services/checkWebsiteStatus.js >> /path/to/logfile.log 2>&1


2. Create another docker container that contains the cron job and have it run all the time, which is more complex to 1st option.

what do you think?

Reply:
I agree that option 1 is better. Maybe take a look at our [site installation directions](https://openenergydashboard.org/helpV1_0_0/adminInstallation/) starting at step 10 to see how we already set up for cron jobs. OED provides a cron script that the site edits and then uses for cron. Let me know if you think that is good for this situation.

Reply:
I just took a look and it seems like they are the cron command in a bash file to simplify the process of running the script. One thing I dont understand is, take sendLogEmailCron.bash for example, the comment says 'This should be copied to /etc/cron.daily/'

So is there such path as '/etc/cron.daily/' in the codespace? i couldnt seem to find it

Reply:
That is in the Linux/Unix file system and not in the OED files/container. It is a common place where all scripts in that directory are automatically run at the frequency specified by that directory. In the one you gave it is every day. There is also an hourly directory. Using this guarantees it runs whenever the server is up (even after a restart). Help?

Reply:
Ah ok i see. Should i create a new bash file and reuse the command and only change the file name then?

sendLogEmail

For example, I will replace 'sendLogEmail' with 'checkWebsiteStatus'. Of couse 'checkWebsiteStatus' will be in package.json with the relative path of 'checkWebsiteStatus.js'

Reply:
I think that is correct (I can verify when I see it). When you say replace I assume you mean you have a copy of one into a new file that you edit and leave the other one untouched.

Reply:
Correct, I am creating a new file and leaving the other one untouched :)